src/inventories/batches/InventoryBatchCreationService.ts

```diff
@@ -0,0 +1,60 @@
+import * as Q from 'q';
+import * as uuidV4 from 'uuid/v4';
+import { AxiosRequestConfig } from 'axios';
+
+import { Logger, LogLevel } from 'channelape-logger';
+import LocationsService from '../../locations/service/LocationsService';
+import BatchCreationRequest from '../quantities/model/AdjustmentRequest';
+import Adjustment from '../quantities/model/Adjustment';
+import Resource from '../../model/Resource';
+import Version from '../../model/Version';
+import RequestClientWrapper from '../../RequestClientWrapper';
+import RestService from '../../service/RestService';
+
+export class InventoryBatchCreationService extends RestService {
+  private EXPECTED_POST_STATUS: number = 201;
+
+  private readonly logger = new Logger('InventoryBatchCreationService', LogLevel.VERBOSE);
+
+  constructor(
+    private readonly client: RequestClientWrapper,
+    private readonly locationsService: LocationsService
+  ) {
+    super();
+  }
+
+  private async getBusinessIdFromLocation(locationId: string): Promise<string> {
+    try {
+      const location = await this.locationsService.get(locationId);
+      return location.businessId;
+    } catch (error) {
+      this.logger.error(error);
+      throw new Error(`Failed to retrieve business ID from location: ${locationId}`);
+    }
+  }
+
+  private static generateIdempotentKey(
+    inventoryItemId: string,
+    status: string,
+    locationId: string,
+    deduplicationKey: string
+  ): string {
+    return `${deduplicationKey}_${locationId}_${inventoryItemId}_${status}`;
+  }
+
+  private createBatchInventory(batchCreationRequest: BatchCreationRequest) {
```

Comment:
Just call this createBatch. Also is there a reason we don't have a public method here? You won't be able to call anything here.